internal/resourcestatus/resourcekey.go

```diff
@@ -172,7 +173,10 @@ func aggregateResourceStatesClustersMap(resourceKeyStates resourceStatesByResour
 				resource.IncompleteState = true
 			}
 
-			appendToPerClusterState(&resource.PerClusterState, entry.state, entry.clusterID)
+			if size < 1024*1024 { // 1mb
```

Comment:
Please consider truncating the result instead. The `resourceKeyStates` map will be iterated in a random order, producing a different result on each run. This could cause an infinite loop of status updates, trying to reconcile it to a value that changes with every execution